sort/cocktailsort.go

```diff
@@ -0,0 +1,54 @@
+// Implementation of Cocktail sorting
+// reference: https://en.wikipedia.org/wiki/Cocktail_shaker_sort
+
+package sort
+
+import "github.com/TheAlgorithms/Go/constraints"
+
+// Cocktail sort is a variation of bubble sort, operating in two directions (beginning to end, end to beginning)
+func Cocktail[T constraints.Integer](arr []T) []T {
```

Comment:
From my understanding, isn't this algorithm applicable to all `Ordered` types?

Reply:
Right.